src/core/internal/array/appending.d

```diff
@@ -76,79 +76,39 @@ template _d_arrayappendcTXImpl(Tarr : T[], T)
     alias _d_arrayappendcTXTrace = _d_HookTraceImpl!(Tarr, _d_arrayappendcTX, errorMessage);
 }
 
-/// Implementation of `_d_arrayappendT` and `_d_arrayappendTTrace`
-template _d_arrayappendTImpl(Tarr : T[], T)
+/// Implementation of `_d_arrayappendT`
+ref Tarr _d_arrayappendT(Tarr : T[], T)(return ref scope Tarr x, scope Tarr y) @trusted
 {
-    import core.internal.array.utils : _d_HookTraceImpl;
-
-    private enum errorMessage = "Cannot append to array if compiling without support for runtime type information!";
-
-    /**
-     * Append array `y` to array `x`.
-     * Params:
-     *  x = what array to append to, taken as a reference
-     *  y = what should be appended
-     * Returns:
-     *  The new value of `x`
-     * Bugs:
-    *   This function template was ported from a much older runtime hook that bypassed safety,
-    *   purity, and throwabilty checks. To prevent breaking existing code, this function template
-    *   is temporarily declared `@trusted pure` until the implementation can be brought up to modern D expectations.
-     */
-    static if (isCopyingNothrow!T)
-        ref Tarr _d_arrayappendT(return ref scope Tarr x, scope Tarr y) @trusted pure nothrow
-        {
-            pragma(inline, false);
-
-            mixin(_d_arrayappendTBody);
-        }
-    else
-        ref Tarr _d_arrayappendT(return ref scope Tarr x, scope Tarr y) @trusted pure
-        {
-            pragma(inline, false);
+    pragma(inline, false);
 
-            mixin(_d_arrayappendTBody);
-        }
+    import core.stdc.string : memcpy;
+    import core.internal.traits : hasElaborateCopyConstructor, Unqual;
+    import core.lifetime : copyEmplace;
 
-    private enum _d_arrayappendTBody = q{
-        import core.stdc.string : memcpy;
-        import core.internal.traits : hasElaborateCopyConstructor, Unqual;
-        import core.lifetime : copyEmplace;
+    auto length = x.length;
 
-        auto length = x.length;
+    _d_arrayappendcTXImpl!Tarr._d_arrayappendcTX(x, y.length);
 
-        _d_arrayappendcTXImpl!Tarr._d_arrayappendcTX(x, y.length);
-
-        static if (hasElaborateCopyConstructor!T)
-        {
-            foreach (i; 0 .. y.length)
-                copyEmplace(y[i], x[length + i]);
-        }
-        else
-        {
-            // blit all elements at once
-            if (y.length)
-                memcpy(cast(Unqual!T *)&x[length], cast(Unqual!T *)&y[0], y.length * T.sizeof);
-        }
-
-        return x;
-    };
+    static if (hasElaborateCopyConstructor!T)
+    {
+        foreach (i; 0 .. y.length)
```

Comment:
I'd suggest to rewrite this loop

```d
foreach (i, ref e; y)
     copyEmplace(e, x[length + i]);
```

to eliminate the useless bound check on `y[i]`

Reply:
or same loop but `y.ptr[i]`